### PR TITLE
Minor grammatical improvements

### PR DESCRIPTION
**Pull request by NicholasSterling**
This only covers about the first 20% of the file; I want to see how the changes are received.

And thanks for writing this guide!

### CODE REVIEW

**Comment by teh-cmc** [comment]
> I want to see how the changes are received

Any kind of improvement is def welcome :)
I'll have a look when I get a minute (and some sleep).

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## Who is this book for?
 
-This book is targeted towards experienced programmers that already feel somewhat comfortable with vanilla Rust (you definitely do not need to be an "expert" though, I certainly am not) and would like to dip their toes into its async ecosystem.
+This book is targeted towards experienced programmers that already feel somewhat comfortable with vanilla Rust (you definitely do not need to be an "expert," though; I certainly am not) and would like to dip their toes into its async ecosystem.
```
Comment (NicholasSterling) [inline README.md:7]: I'll leave more info in comments.

\<independent clause\>, \<independent clause\>
is a "comma splice."

File: README.md [hunk 2]
```diff
@@ -61,25 +61,25 @@ rustc-version: rustc 1.40.0-nightly (b520af6fd 2019-11-03)
 
 So what's the deal here? Why do we need asynchronous Rust in the first place?
 
-In my experience, more often than not, the first answers to come up are along the lines of "to workaround the limitations of synchronous I/O".  
+In my experience, more often than not the first answers to come up are along the lines of "to work around the limitations of synchronous I/O".
```
Comment (NicholasSterling) [inline README.md:64]: "workaround" is the noun, the thing you create when you work around something.

File: README.md [hunk 3]
```diff
@@ -61,25 +61,25 @@ rustc-version: rustc 1.40.0-nightly (b520af6fd 2019-11-03)
...
+Hence in this first chapter, we'll do just that -- demystify iterators and closures:
 - What they are and what they're not.
 - Their memory representation and execution model.
 - Their interesting similarities.
 - And last but not least: why would we need and/or want something else to begin with?
 
 ## 1.1. Iterators
 
-An iterator is a state-machine that you can move forward by calling its `.next` method.  
+An iterator is a state machine that you can move forward by calling its `.next` method.
```
Comment (NicholasSterling) [inline README.md:82]: \<noun\> \<noun\> where the first effectively acts as an adjective, telling "what kind" of the second
wedding ring
government spending
state machine
object orientation

Note that in "object-oriented programming" we have a noun-adjective pair acting as an adjective, which requires a hyphen.

File: README.md [hunk 4]
```diff
@@ -142,11 +142,11 @@ for i in Range::new(1usize, 4, 1) {
...
 ```
 
-Straighforward stuff. This does demonstrate a couple important characteristics of iterators, though.
+Straightforward stuff. This does demonstrate a couple of important characteristics of iterators, though.
 
 **Laziness**
 
-Iterators are lazy, they don't do anything unless _polled_.  
+Iterators are lazy; they don't do anything unless _polled_.  
 In this case the for loop is doing the polling, as it desugars to something along these lines:
```
Comment (NicholasSterling) [inline README.md:150]: Another comma splice.

File: README.md [hunk 5]
```diff
@@ -164,11 +164,11 @@ Similarly, one can effectively _cancel_ the execution of an iterator at any time
 Our iterator is nothing more than a vanilla structure (its state) with a `.next` method defined on it.  
 In this example, the iterator sits on `main`'s stack, and answers to the same rules as any other struct: ownership, lifetimes, marker traits, etc.
 
-There really isn't any kind of magic going on here: no hidden allocations, no codegen, no nothing; it's as dull as it gets.
+There really isn't any kind of magic going on here: no hidden allocations, no codegen, no nothing.  It's as dull as it gets.
```
Comment (NicholasSterling) [inline README.md:167]: The colon implies that what follows elucidates what precedes it.  That's true only up to the semicolon.

https://painintheenglish.com/case/616

File: README.md [hunk 6]
```diff
@@ -273,27 +273,27 @@ assert_eq!(None, it.next());
...
 # [...]
 `Bounds<Bounds<Bounds<Range<usize>, usize>, usize>, usize>`: 72 bytes, alignment: 8 bytes
 ```
-How could it? It's just blindly monomorphizing structs inside of other structs, that's all there is to it!
+How could it? It's just blindly monomorphizing structs inside of other structs; that's all there is to it!
 
 **What we have so far**
 
-We can now build up complex state-machines by composing them from smaller, simpler parts... and what's more, we can even do it in an expressive, readable and maintainable way.
+We can now build up complex state machines by composing them from smaller, simpler parts... and what's more, we can even do it in an expressive, readable and maintainable way.
```
Comment (NicholasSterling) [inline README.md:286]: I didn't make the change, but I personally prefer to use the "Oxford comma":
... in an expressive, readable, and maintainable way.

Dealer's choice, though.

File: README.md (hunk 4 above)
Comment (NicholasSterling) [reply README.md:150]: That could also be a colon or a dash, and perhaps should be, since the right elucidates the left.

File: README.md (hunk 1 above)
Comment (teh-cmc) [reply README.md:7]: I am extremely disturbed by that comma being inside the quotes: `"expert,"`.
Is that normal..?

**Comment by teh-cmc** [comment]
Thanks for the explanations!
Apart from that first comma (see inline comment above), it all looks good to me.

Should I merge this or do you feel like going over the rest of the chapter?

File: README.md (hunk 1 above)
Comment (bruceadams) [reply README.md:7]: Yes. Putting the comma inside the quote in normal. Reading https://www.quickanddirtytips.com/education/grammar/how-to-use-quotation-marks suggests that it is standard in the US, but optional elsewhere (which I didn't know until now).

Comment (NicholasSterling) [reply README.md:7]: You know, I actually *hate* that that's correct.  BLECH!  And I refuse to do it when communicating what to type, as in

   At the prompt, type "rm foo."

But elsewhere, I grudgingly follow the rules.  Honestly, if you don't want that one, I don't blame you.  Let me know.  Or actually, I remember checking some box that I *think* allows you to edit my version, so feel free.

**Comment by NicholasSterling** [comment]
Why don't we do them piecemeal -- that way these old changes that you've already evaluated will be out of the way when you get the next batch.

**Comment by teh-cmc** [comment]
Ouch, I've just realized that you've modified the autogenerated/rendered text rather than the source :(

I'll port your changes to the source files, remove the comma thingy, then it's a merge.

**Comment by NicholasSterling** [comment]
Oh, sorry!  I am used to software projects where src/ contains code and the README is maintained by hand.  It never occurred to me that in spite of being version-controlled, the README might be generated from /src.

I'm fine with making the changes against the src/ files if you prefer that I do it.  Again, sorry.

**Comment by teh-cmc** [comment]
> I'm fine with making the changes against the src/ files if you prefer that I do it. Again, sorry.

As you wish. I'll do it at the same time I write chapter 2, if it hasn't been already done then. No worries :)

**Comment by NicholasSterling** [comment]
Getting rid of this; I have made the changes against the src/ files.

**Pull request closed without merging**